doc/sphinx/proof-engine/ltac.rst

```diff
@@ -88,91 +88,72 @@ mode but it can also be used in toplevel definitions as shown below.
-                     : type of `term`
-                     : constr : `term`
-                     : uconstr : `term`
-                     : type_term `term`
-                     : numgoals
-                     : guard `test`
-                     : assert_fails `ltac_expr3`
-                     : assert_succeeds `ltac_expr3`
-                     : `tactic`
-                     : `qualid` `tacarg` ... `tacarg`
-                     : `atom`
-   atom              : `qualid`
-                     : ()
-                     : `int`
-                     : ( `ltac_expr` )
-   component : `string` | `qualid`
-   message_token     : `string` | `ident` | `int`
-   tacarg            : `qualid`
-                     : ()
-                     : ltac : `atom`
-                     : `term`
-   let_clause        : `ident` [`name` ... `name`] := `ltac_expr`
-   context_rule      : `context_hyp`, ..., `context_hyp` |- `cpattern` => `ltac_expr`
-                     : `cpattern` => `ltac_expr`
-                     : |- `cpattern` => `ltac_expr`
-                     : _ => `ltac_expr`
-   context_hyp       : `name` : `cpattern`
-                     : `name` := `cpattern` [: `cpattern`]
-   match_rule        : `cpattern` => `ltac_expr`
-                     : context [`ident`] [ `cpattern` ] => `ltac_expr`
-                     : _ => `ltac_expr`
-   test              : `int` = `int`
-                     : `int` (< | <= | > | >=) `int`
-   selector          : [`ident`]
-                     : `int`
-                     : (`int` | `int` - `int`), ..., (`int` | `int` - `int`)
-   toplevel_selector : `selector`
-                     : all
-                     : par
-                     : !
-
-.. productionlist:: coq
-   top              : [Local] Ltac `ltac_def` with ... with `ltac_def`
-   ltac_def         : `ident` [`ident` ... `ident`] := `ltac_expr`
-                    : `qualid` [`ident` ... `ident`] ::= `ltac_expr`
+.. insertprodn ltac_expr tactic_atom
+
+.. prodn::
+   ltac_expr ::= @binder_tactic
+   | @ltac_expr4
+   binder_tactic ::= fun {+ @name } => @ltac_expr
+   | let {? rec } @let_clause {* with @let_clause } in @ltac_expr
+   | info @ltac_expr
+   let_clause ::= @name := @ltac_expr
+   | @ident {+ @name } := @ltac_expr
+   ltac_expr4 ::= @ltac_expr3 ; @binder_tactic
+   | @ltac_expr3 ; @ltac_expr3
+   | @ltac_expr3 ; [ {? @multi_goal_tactics } ]
+   | @ltac_expr3 ; [> {? @multi_goal_tactics } ]
+   | @ltac_expr3
+   multi_goal_tactics ::= {? @ltac_expr } %| @multi_goal_tactics
+   | {? @ltac_expr } .. {? %| {*| {? @ltac_expr } } }
+   | @ltac_expr
+   ltac_expr3 ::= try @ltac_expr3
+   | do @int_or_var @ltac_expr3
+   | timeout @int_or_var @ltac_expr3
+   | time {? @string } @ltac_expr3
+   | repeat @ltac_expr3
+   | progress @ltac_expr3
+   | once @ltac_expr3
+   | exactly_once @ltac_expr3
+   | infoH @ltac_expr3
+   | abstract @ltac_expr2
+   | abstract @ltac_expr2 using @ident
+   | only @selector : @ltac_expr3
```

Comment:
These could maybe be extracted as a `ltac_tactical` or `ltac_combinator` non-terminal. We might also decide to extract only `try`, `do`, `repeat`, `progress`, `once`, `exactly_once` together because they are close from a semantic point of view and it makes sense that they are documented in a single section.

Reply:
The grouping under these nonterminals is not be obvious if you don't already know the constructs well.

How do you make these constructs easily discoverable without reading the entire chapter?  Or easy to find in the doc; text search in the browser may give a lot of false positives.  Introductions will help but I think indexes are part of solution.